libr/include/r_util/pj.h

```diff
@@ -14,10 +14,24 @@ typedef struct pj_t {
 	bool is_key;
 	char braces[R_PRINT_JSON_DEPTH_LIMIT];
 	int level;
+	int str_encoding;
+	int num_encoding;
 } PJ;
 
+/* new encoding options of j commands */
+enum {
+	PJ_ENCODING_DEFAULT = 0,
+	PJ_ENCODING_STR_BASE64,
+	PJ_ENCODING_STR_HEX,
+	PJ_ENCODING_STR_ARRAY,
+	PJ_ENCODING_STR_STRIP,
+	PJ_ENCODING_NUM_STR,
+	PJ_ENCODING_NUM_HEX
+};
+
 /* lifecycle */
 R_API PJ *pj_new(void);
+R_API PJ *pj_new_with_encoding(const char *str_encoding, const char *num_encoding);
```

Comment:
from an API point of view I would just pass the enum value instead of strings. Let `RCore` deal with converting strings to enum values. Also, give a name to the enum above and maybe split it in `enum PJEncodingStr` and `enum PJEncodingNum`.

Reply:
Ok, pushing the change in a second.